0miner

```diff
@@ -149,68 +154,76 @@ fi
 if [[ $ALGO == ETHASH ]]
 then
 ## Declare Miner Start
+  miner_opts="$(dgh_enabled_devices "${MDPA[$ETHASH_MINER]}" "${MDDA[$ETHASH_MINER]}" "${MDTA[$ETHASH_MINER]}" "$(dgh_get_miner_opts "$ETHASH_MINER" "$COIN" "$ALGO")")"
```

Comment:
if you change this line as follows you can put is once before the "if ALGO ==" tree:
```
miner_opts="$(dgh_enabled_devices "${MDPA[${!xminer}]}" "${MDDA[${!xminer}]}" "${MDTA[${!xminer}]}" "$(dgh_get_miner_opts "${!xminer}" "$COIN" "$ALGO")")" 
```

Reply:
Actually dgh_get_miner_opts is capable of dyanmic look up which GLOBAL variable to be used by passing it $COIN and $ALGO, i'm restricting the use of it in 0miner because of how 0miner is building the command. Don't wish to introduce more head spinning.
IF you look at the description of dgh_get_miner_opts, without passing in the miner value, it will check GLOBAL VARIABLE in the ORDER of ${COIN}_MINER, ${ALGO}_MINER to retrieve the defined $miner, then check in the order of ${COIN}_${miner}_OPTS, ${ALGO}_${miner}_OPTS, ${miner}_OPTS to try to figure out which OPTS to be used, it's built this way having in mine that you can assign different GPU for running different ALGO/COIN, which I think will prepare you guys for multi miner implementation

Reply:
ETHASH_MINER is "hardcoded in the ETHASH if condition is because I'm afraid to introduce too much flexibity and cause hard to test bugs in 0 miner

Reply:
I've looked some parts of your implementation, that's why I say you can simply assign miner_opts outside that if block. You have to take into account that the entire "if ALGO ==" block is likely to be dropped as soon as builtin miners get deprecated. If you need the miner name that is into ${!xminer}, will be ETHMINER or whatever when ALGO==ETHASH and COIN=ETH. Obviously the implementation should be transparent: meanwhile you have no support for general miners, you have to handle unsupported miners as well by falling back to some predefined behavior, like all gpus enabled for example.
To support a general mniner you would take as input something that describes the miner option argument format, not the miner name, but for now it is ok not to have such a mechanism, so take the miner name from ${!xminer}

Reply:
I have moved  miner_opts out from ETHASH, ZHASH and EQUIHASH condition. However, the algo using CCMINER remained, due to how it's defined. IMHO we should leave it like that for the time being I'm reluctant to spend too much time making assumption. I will revisit it when it's clearer on what changes will be made to the miners related  implementation down the road. What do you think?

Just to clarify,
My intention is not to have the disabled-gpu script aware of each miner's information, but a  function to parse and manipulate string based on arguments passed in related global variable defined in 1bash. The script is made not to find the dev prefix and GPU # format like how it can find the opts, because I think those can get defined in the json file once your portable miner is implemented.
The global variable MDPA, MDDA, MDTA should be removed after built in miners get deprecated, they are just there for the transition period.

Reply:
"you have to handle unsupported miners as well by falling back to some predefined behavior, like all gpus enabled for example."

There's really no fall back for unknown miners, this totally depends on what the miner accept for it to work. if we don't know a miner, making assumption of the prefix might even break the miner, might as well do nothing :)